distribution/integration/test-common/integration-test-utils/src/main/java/org/wso2/carbon/integration/common/utils/CarbonCommandToolsUtil.java

```diff
@@ -0,0 +1,388 @@
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.extensions.ExtensionConstants;
+import org.wso2.carbon.automation.extensions.servers.carbonserver.CarbonServerManager;
+import org.wso2.carbon.automation.extensions.servers.utils.ClientConnectionUtil;
+import org.wso2.carbon.automation.extensions.servers.utils.InputStreamHandler;
+import org.wso2.carbon.automation.extensions.servers.utils.ServerLogReader;
+import org.wso2.carbon.integration.common.exception.CarbonToolsIntegrationTestException;
+import org.wso2.carbon.server.admin.ui.ServerAdminClient;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+
+public class CarbonCommandToolsUtil {
+
+    /**
+     * This class has the method which using by carbon tools test cases
+     */
+
+    private static final Log log = LogFactory.getLog(CarbonCommandToolsUtil.class);
+    private static int TIMEOUT = 180 * 1000; // Max time to wait
+    private static String carbonHome = null;
+    private static final long DEFAULT_START_STOP_WAIT_MS = 1000 * 60 * 5;
+    private static ServerLogReader inputStreamHandler;
+    private static final String SERVER_STARTUP_MESSAGE = "Mgt Console URL";
+
+    /**
+     * This method is to execute commands and reading the logs to find the expected string.
+     *
+     * @param directory      - Directory which has the file to be executed .
+     * @param cmdArray       - Command array to be executed.
+     * @param expectedString - Expected string in  the log.
+     * @return boolean - true : Found the expected string , false : not found the expected string.
+     * @throws CarbonToolsIntegrationTestException - Error while getting the command directory
+     */
+    public static boolean isScriptRunSuccessfully(String directory, String[] cmdArray,
+                                                  String expectedString)
+            throws CarbonToolsIntegrationTestException {
+        boolean isFoundTheMessage = false;
+        BufferedReader br = null;
+        Process process = null;
+        try {
+            File commandDir = new File(directory);
+            process = Runtime.getRuntime().exec(cmdArray, null, commandDir);
+            String line;
+            long startTime = System.currentTimeMillis();
+            while ((System.currentTimeMillis() - startTime) < TIMEOUT) {
+                br = new BufferedReader(new InputStreamReader(process.getInputStream(), "UTF-8"));
+                while ((line = br.readLine()) != null) {
+                    log.info(line);
+                    if (line.contains(expectedString)) {
+                        log.info("found the string " + expectedString + " in line " + line);
+                        isFoundTheMessage = true;
+                        break;
+                    }
+                }
+                if (isFoundTheMessage) {
```

Comment:
add the isFoundTheMessage condition to while loop. You can break the loop when it set to true)